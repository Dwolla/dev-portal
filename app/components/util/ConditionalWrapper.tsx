// Used for conditionally wrapping elements

const ConditionalWrapper = ({ condition, wrapper, children }) =>
  condition ? wrapper(children) : children;

export default ConditionalWrapper;
