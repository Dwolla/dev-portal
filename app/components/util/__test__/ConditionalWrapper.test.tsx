import renderer from "react-test-renderer";
import ConditionalWrapper from "../ConditionalWrapper";

test("ContionalWrapper", () => {
  const tree = renderer
    .create(
      <ConditionalWrapper
        condition="true"
        wrapper={(children) => <div>{children}</div>}
      >
        <div></div>
      </ConditionalWrapper>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
