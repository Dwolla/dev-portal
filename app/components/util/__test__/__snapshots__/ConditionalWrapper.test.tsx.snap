// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContionalWrapper 1`] = `
<div>
  <div />
</div>
`;
